File: question/014LongestCommonPrefix.cpp
//1. Recursion. firstChar+LCP(substr(1)). 4ms
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        if(strs.size()==1) return strs[0];
        if(strs[0].length()==0) return "";

        string res = "";
        char first = strs[0][0];
        
        res += first;
        for(int i=0;i<strs.size();i++){
            if(strs[i][0]!=first) return "";
            strs[i] = strs[i].substr(1);
        }
        
        return res+longestCommonPrefix(strs);
    }
};

File: Algo/Leetcode/014LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        n = len(strs)
        i = 0
        con = True
        while con:
            if i >= len(strs[0]):
                con = False
                i += 1
                break

            t = strs[0][i]
            for k in range(1, n):
                if i >= len(strs[k]) or strs[k][i] != t:
                    con = False
                    break
            i += 1

        return strs[0][:max(0, i-1)] 